examples/fungible-token/tests/general.rs

```diff
@@ -0,0 +1,83 @@
+use near_sdk::json_types::U128;
+use near_sdk::serde_json::json;
+use near_sdk_sim::test_runtime::{init_test_runtime, to_yocto};
+use near_sdk_sim::{transaction::ExecutionOutcome, TestRuntime, User, DEFAULT_GAS, STORAGE_AMOUNT};
+use std::str::FromStr;
+
+extern crate fungible_token;
+use fungible_token::FungibleTokenContract;
+use near_sdk::PendingContractTx;
+
+near_sdk_sim::lazy_static::lazy_static! {
+    static ref TOKEN_WASM_BYTES: &'static [u8] = include_bytes!("../res/fungible_token.wasm").as_ref();
+}
+
+fn init(initial_balance: u128) -> (TestRuntime, FungibleTokenContract, User) {
+    let runtime = init_test_runtime(None);
+    let root = runtime.get_root();
+    // let balance: U128 = initial_balance.into();
+    let contract = FungibleTokenContract::_new("contract".to_string());
+    let contract_user = root.deploy_and_init(
+        &TOKEN_WASM_BYTES,
+        contract.new(root.account_id.clone(), initial_balance.into()),
+    );
+    let alice = runtime.create_user("alice".to_string(), to_yocto("100"));
+    (runtime, contract, alice)
+}
```

Comment:
Things I like about the stuff inside `init`:

* No unnecessary ceremony with Rust unwrapping, options, or error returns
* `deploy_and_init` matches the concept from near-cli
* Preserving function calls as-written, with `contract.fn_name(...args)`, vs something like `contract.call("fn_name", ...args)`

Some mostly-syntactic things I find confusing:

* needing to pass `None` into `init_test_runtime` even when you don't want to change any settings. If we make this a macro could we avoid that?
* "root" instead of matching our currently-more-common name, `master_account`
* the need to initialize `contract`
* having `WhateverContract::_new` be the way to initialize a contract
* having both `contract` and `contract_user`
* the word "user" being present at all, since NEAR vernacular usually only refers to "accounts"
* what is `&TOKEN_WASM_BYTES`?
* why do we need to "deploy" the contract?
* the difference in how an "account with a contract" is initialized vs how an account without one is initialized

So let me play with some possible alternative syntax.

```rust
fn init(initial_balance: u128) -> (TestRuntime, FungibleTokenContract, User) {
    let runtime = init_test_runtime();
    let master_account = runtime.get_master_account();
    let contract = create_account!({
        id: "contract".to_string(),
        balance: initial_balance.into(),
        contract: FungibleTokenContract,
    });
    // if we use a macro, can we return different types for "AccountWithContract" and "AccountWithoutContract"?
    let alice = create_account!({
        id: "alice".to_string(),
        balance: to_yocto("100"),
    });
    master_account.call(contract.new()); // this loses deploy_and_init, but maybe that's ok?
    let contract = FungibleTokenContract::_new("contract".to_string());
    (runtime, contract, alice)
}
```

Reply:
Or maybe:

```rust
fn init(initial_balance: u128) -> (TestRuntime, FungibleTokenContract, User) {
    let runtime = init_test_runtime!();
    let master_account = runtime.get_master_account();
    let alice = master_account.create_account("alice".to_string(), to_yocto("100"));
    let contract = master_account.create_account("contract".to_string(), initial_balance.into());
    let contract = master_account.deploy(contract, FungibleTokenContract);
    master_account.call(contract.new(master_account.id.clone(), initial_balance.into()));
    (runtime, contract, alice)
}